src/twisted/mail/imap4.py

```diff
@@ -209,33 +275,76 @@ def clean(self):
         Clean ranges list, combining adjacent ranges
         """
 
-        self.ranges.sort()
+        ranges = sorted((self._swap(low, self._infinity, ifIs=None),
+                         self._swap(high, self._infinity, ifIs=None))
+                         for low, high in self.ranges)
 
-        oldl, oldh = None, None
-        for i,(l, h) in enumerate(self.ranges):
-            if l is None:
+        mergedRanges = [(float('-inf'), float('-inf'))]
```

Comment:
It looks like `inf`, `-inf`, and `nan` became portable [a while ago](https://bugs.python.org/issue1635):

> The float() function will now turn the string nan into an IEEE 754 Not A Number value, and +inf and -inf into positive or negative infinity. **This works on any platform with IEEE 754 semantics**. (Contributed by Christian Heimes; issue 1635.)

(from https://docs.python.org/2/whatsnew/2.6.html, emphasis added.)

So I think the infinities are safe...